feature_engine/creation/cyclical.py

```diff
@@ -0,0 +1,140 @@
+
+    References
+    ----------
+    http://blog.davidkaleko.com/feature-engineering-cyclical-features.html
+    """
+
+    def __init__(
+            self, variables: Union[None, int, str, List[Union[str, int]]] = None,
+            max_values: Optional[Dict[str, Union[int, float]]] = None,
+            drop_original: Optional[bool] = False
+    ) -> None:
+        self.variables = _check_input_parameter_variables(variables)
+        self.max_values = max_values
+        self.drop_original = drop_original
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        Learns the max_value of each of the numerical variables.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The training input samples. Can be the entire dataframe, not just the
+            variables to transform.
+
+        y : pandas Series, default=None
+            It is not needed in this transformer. You can pass y or None.
+
+        Raises
+        ------
+        TypeError
+            If the input is not a Pandas DataFrame.
+        ValueError:
+            - If some of the columns contains NaNs.
+            - If some of the mapping keys are not present in variables.
+
+        Returns
+        -------
+        self
+        """
+
+        # check input dataframe
+        X = super().fit(X)
+        _check_contains_na(X, self.variables)
+
+        if self.max_values is None:
+            self.max_values_ = X[self.variables].max().to_dict()
+        else:
+            for key in list(self.max_values.keys()):
+                if key not in self.variables:
+                    raise ValueError(f'The mapping key {key} is not present'
+                                     f' in variables.')
+            self.max_values_ = self.max_values
+        self.input_shape_ = X.shape
+        return self
+
+    def transform(self, X: pd.DataFrame):
+        """
+        Apply a Ciclycal transformation.
+
+        Parameters
+        ----------
+        X : Pandas DataFrame of shame = [n_samples, n_features]
+            The data to be transformed.
+
+        Raises
+        ------
+        TypeError
+            If the input is not Pandas DataFrame.
+
+        Returns
+        -------
+        X : Pandas dataframe.
+            Depending if drop_original was set to True the dataframe will have
+            new columns which are twice the amount of columns in variables. If
+            set to False it will have twice the amount of columns in variables
+            and the original columns.
+        """
+        X = super().transform(X)
+
+        for variable in self.variables:
+            max_value = self.max_values_[variable]
+            X[f'{variable}_sin'] = np.sin(X[variable] * (2. * np.pi / max_value))
+            X[f'{variable}_cos'] = np.cos(X[variable] * (2. * np.pi / max_value))
+            if self.drop_original:
+                X.drop(columns=variable, inplace=True)
+
```

Comment:
could we remove the indentation and remove all variables in one go, instead of inside the loop?